src/azure/builder.rs

```diff
@@ -658,8 +658,8 @@ impl MicrosoftAzureBuilder {
         };
 
         match parsed.scheme() {
-            "az" | "adl" | "azure" => self.container_name = Some(validate(host)?),
-            "abfs" | "abfss" => {
+            "adl" | "azure" => self.container_name = Some(validate(host)?),
+            "az" | "abfs" | "abfss" => {
```

Comment:
Can you add a test of the form `az://container@account.dfs.core.windows.net/path-part/file`?

Reply:
Sure. Done.